fabric-rendering-v1/src/main/java/net/fabricmc/fabric/impl/client/rendering/BuiltinItemRendererRegistryImpl.java

```diff
@@ -0,0 +1,31 @@
+package net.fabricmc.fabric.impl.client.rendering;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import net.minecraft.item.Item;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.fabricmc.fabric.api.client.rendering.v1.BuiltinItemRenderer;
+import net.fabricmc.fabric.api.client.rendering.v1.BuiltinItemRendererRegistry;
+
+@Environment(EnvType.CLIENT)
+public enum BuiltinItemRendererRegistryImpl implements BuiltinItemRendererRegistry {
+	INSTANCE;
+
+	private static final Map<Item, BuiltinItemRenderer> RENDERERS = new HashMap<>();
+
+	@Override
+	public void register(Item item, BuiltinItemRenderer renderer) {
+		if (RENDERERS.containsKey(item)) {
+			throw new IllegalArgumentException("Item " + item + " already has a builtin renderer!");
```

Comment:
I would recommend spitting out the Identifier this item is registered under rather than potentially having a jumbled mess.

If it isn't in the registry then obviously throw that as well

Reply:
`Item.toString()` already gets the registry ID, though I guess make it get the ID directly.

Reply:
Hmm if it does already then that is fine

Reply:
I made it use `Registry.getId` anyway because vanilla only returns the path for some reason.